`lfs_config.h` is not used by `Ninja` build system?

I use this component in this environment:

- Windows 10
- VSCode + Espressif IDF Extention
- ESP-IDF v4.4

that uses `Ninja` to build the project.  


To see which file is used in compilation process I added two `#warning` before and after [this line](https://github.com/littlefs-project/littlefs/blob/ead50807f1ca3fdf2da00b77a0ce02651ded2d13/lfs_util.h#L20) in `lfs_util.h`.  
Final code seems like this:

```C
#ifdef LFS_CONFIG
#define LFS_STRINGIZE(x) LFS_STRINGIZE2(x)
#define LFS_STRINGIZE2(x) #x
#include LFS_STRINGIZE(LFS_CONFIG)

#warning "using lfs_config.h"  // <-------

#else

#warning "using lfs_utils.h" // <-------

// System includes
#include <stdint.h>

```

Then building the project by invoking `idf.py build` command in `ESP-IDF Shell` I get these `#warning "using lfs_utils.h"` that indicates `lfs_config.h` is not used.

```Shell
PS> idf.py build
Executing action: all (aliases: build)
Running ninja in directory PATH_TO_PROJECT\PROJECT_NAME\build
Executing "ninja all"...
[0/1] Re-running CMake...
-- ccache will be used for faster recompilation
-- Building ESP-IDF components for target esp32

 .................  [removed]

[1/13] Performing build step for 'bootloader'
[1/1] ...... [removed]
[2/11] Building C object esp-idf/esp_littlefs/CMakeFiles/__idf_esp_littlefs.dir/src/littlefs/lfs_util.c.obj
In file included from ../components/esp_littlefs/src/littlefs/lfs_util.c:7:
../components/esp_littlefs/src/littlefs/lfs_util.h:22:2: warning: #warning "using lfs_utils.h" [-Wcpp]
 #warning "using lfs_utils.h"
  ^~~~~~~
[4/11] Building C object esp-idf/esp_littlefs/CMakeFiles/__idf_esp_littlefs.dir/src/littlefs_api.c.obj
In file included from ../components/esp_littlefs/src/littlefs/lfs.h:12,
                 from ../components/esp_littlefs/src/littlefs_api.c:12:
../components/esp_littlefs/src/littlefs/lfs_util.h:22:2: warning: #warning "using lfs_utils.h" [-Wcpp]
 #warning "using lfs_utils.h"
  ^~~~~~~
[5/11] Building C object esp-idf/esp_littlefs/CMakeFiles/__idf_esp_littlefs.dir/src/esp_littlefs.c.obj
In file included from ../components/esp_littlefs/src/littlefs/lfs.h:12,
                 from ../components/esp_littlefs/include/esp_littlefs.h:21,
                 from ../components/esp_littlefs/src/esp_littlefs.c:41:
../components/esp_littlefs/src/littlefs/lfs_util.h:22:2: warning: #warning "using lfs_utils.h" [-Wcpp]
 #warning "using lfs_utils.h"
  ^~~~~~~
[6/11] Building C object esp-idf/esp_littlefs/CMakeFiles/__idf_esp_littlefs.dir/src/littlefs/lfs.c.obj
In file included from ../components/esp_littlefs/src/littlefs/lfs.h:12,
                 from ../components/esp_littlefs/src/littlefs/lfs.c:7:
../components/esp_littlefs/src/littlefs/lfs_util.h:22:2: warning: #warning "using lfs_utils.h" [-Wcpp]
 #warning "using lfs_utils.h"
  ^~~~~~~
[10/11] Generating binary image from built executable
...
```

I believe the component [`CMakeLists.txt`](https://github.com/joltwallet/esp_littlefs/blob/master/CMakeLists.txt) must be like this:

```
cmake_minimum_required(VERSION 3.10)

file(GLOB SOURCES src/littlefs/*.c)
list(APPEND SOURCES src/esp_littlefs.c src/littlefs_api.c src/lfs_config.c)
idf_component_register(
    SRCS ${SOURCES}
    INCLUDE_DIRS src include
    REQUIRES spi_flash
    PRIV_REQUIRES esptool_py
    )

## use `lfs_config.h`
set_source_files_properties(
    ${SOURCES}
    PROPERTIES COMPILE_FLAGS "-DLFS_CONFIG=lfs_config.h"
)
```

Am I missing something?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,3 +8,8 @@ idf_component_register(
     REQUIRES spi_flash
     PRIV_REQUIRES esptool_py
     )
+
+set_source_files_properties(
+    ${SOURCES}
+    PROPERTIES COMPILE_FLAGS "-DLFS_CONFIG=lfs_config.h"
+)

